drivers/fuel_gauge/axp2101/fuel_gauge_axp2101.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2025, Felix Moessbauer
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Note: The functions that query the raw values via i2c are named similar
+ * to the ones in the reference implementation in the XPowersLib. This also
+ * applies to the register names (defines).
+ *
+ * Datasheet:
+ * https://github.com/lewisxhe/XPowersLib/blob/a7d06b98c1136c8fee7854b1d29a9f012b2aba83/datasheet/AXP2101_Datasheet_V1.0_en.pdf
+ */
+
+#define DT_DRV_COMPAT x_powers_axp2101_fuel_gauge
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/fuel_gauge.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(fuel_gauge_axp2101, CONFIG_FUEL_GAUGE_LOG_LEVEL);
+
+/* clang-format off */
+/* registers */
+#define AXP2101_STATUS1			0x00
+	#define BAT_PRESENT_MASK	BIT(3)
+#define AXP2101_CHARGE_GAUGE_WDT_CTRL	0x18
+	#define GAUGE_ENABLE_MASK	BIT(3)
+#define AXP2101_ADC_DATA_VBAT_H		0x34
+	#define GAUGE_VBAT_H_MASK	0x1F
+#define AXP2101_ADC_DATA_VBAT_L		0x35
+#define AXP2101_BAT_DET_CTRL		0x68
+	#define BAT_TYPE_DET_MASK	BIT(0)
+#define AXP2101_BAT_PERCENT_DATA	0xA4
+
+/* internal feature flags */
+#define GAUGE_FEATURE_BAT_DET		BIT(0)
+#define GAUGE_FEATURE_GAUGE		BIT(1)
+#define GAUGE_FEATURE_ALL		(GAUGE_FEATURE_BAT_DET | GAUGE_FEATURE_GAUGE)
+/* clang-format on */
+
+struct axp2101_config {
+	struct i2c_dt_spec i2c;
+};
+
+struct axp2101_data {
+	uint8_t features;
+};
+
+static int enable_fuel_gauge(const struct device *dev)
+{
+	const struct axp2101_config *cfg = dev->config;
+	struct axp2101_data *data = dev->data;
+	int ret = 0;
+
+	ret = i2c_reg_update_byte_dt(&cfg->i2c, AXP2101_CHARGE_GAUGE_WDT_CTRL, GAUGE_ENABLE_MASK,
+				     GAUGE_ENABLE_MASK);
+	if (ret < 0) {
+		data->features &= ~GAUGE_FEATURE_GAUGE;
+	}
+	return ret;
```

Comment:
Although it is not blocking, I think Zephyr has many codes in that put a blank line before return. Please consider it if you like it.